components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java

```diff
@@ -71,22 +71,21 @@
     public String jwks() {
 
         String tenantDomain = getTenantDomain();
-        String keystorePath = CarbonUtils.getServerConfiguration().getFirstProperty(SECURITY_KEY_STORE_LOCATION);
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+
+        try {
+            FrameworkUtils.startTenantFlow(tenantDomain);
 
-        try (FileInputStream file = new FileInputStream(keystorePath)) {
-            int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
             final KeyStore keystore;
             Map<String, Certificate> certificatesWithAliases = new HashMap<>();
             if (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equalsIgnoreCase(tenantDomain)) {
+                String keystorePath = CarbonUtils.getServerConfiguration()
+                        .getFirstProperty(SECURITY_KEY_STORE_LOCATION);
                 keystore = KeyStore.getInstance(KeyStore.getDefaultType());
                 String password = CarbonUtils.getServerConfiguration().getFirstProperty(SECURITY_KEY_STORE_PW);
+                FileInputStream file = new FileInputStream(keystorePath);
```

Comment:
Better to do this within a try with resource block to make sure FileInputStream closed

Reply:
With another `try` inside the main try block?

Reply:
Or are you suggesting breaking down the try block into several separate try-catch blocks?

Reply:
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html